framework/quota/test/org/apache/cloudstack/quota/QuotaStatementTest.java

```diff
@@ -0,0 +1,248 @@
+
+        //get last day of the previous month
+        aCalendar = Calendar.getInstance();
+        aCalendar.set(Calendar.MONTH, Calendar.DECEMBER);
+        aCalendar.set(Calendar.DATE, aCalendar.getActualMaximum(Calendar.DAY_OF_MONTH) + 1);
+        assertTrue(" period[1].get(Calendar.DATE) == aCalendar.get(Calendar.DATE)" + period[1].toString(), period[1].get(Calendar.DATE) == aCalendar.get(Calendar.DATE));
+        assertTrue("period[1].get(Calendar.MONTH) == aCalendar.get(Calendar.MONTH)" + period[1].toString(), period[1].get(Calendar.MONTH) == aCalendar.get(Calendar.MONTH));
+
+    }
+
+    @Test
+    public void testStatementPeriodYEARLY() {
+        Calendar date = Calendar.getInstance();
+        Calendar aCalendar = Calendar.getInstance();
+
+        //QUATERLY
+        date = Calendar.getInstance();
+        date.set(Calendar.MONTH, Calendar.JANUARY); // 1 Jan
+        date.set(Calendar.DATE, 1);
+        Calendar period[] = quotaStatement.statementTime(date, STATEMENT_PERIODS.YEARLY);
+        assertTrue("period != null", period != null);
+        assertTrue("period[0].before(period[1])" + period[0].toString(), period[0].before(period[1]));
+        assertTrue("period[0].get(Calendar.DATE) == 1" + period[0].toString(), period[0].get(Calendar.DATE) == 1);
+        assertTrue("period[0].get(Calendar.MONTH) == Calendar.JANUARY" + period[0].toString(), period[0].get(Calendar.MONTH) == Calendar.JANUARY); //january
+
+        //get last day of the previous month
+        aCalendar = Calendar.getInstance();
+        aCalendar.set(Calendar.MONTH, Calendar.DECEMBER);
+        aCalendar.set(Calendar.DATE, aCalendar.getActualMaximum(Calendar.DAY_OF_MONTH) + 1);
+        assertTrue(" period[1].get(Calendar.DATE) == aCalendar.get(Calendar.DATE)" + period[1].toString(), period[1].get(Calendar.DATE) == aCalendar.get(Calendar.DATE));
+        assertTrue("period[1].get(Calendar.MONTH) == aCalendar.get(Calendar.MONTH)" + period[1].toString(), period[1].get(Calendar.MONTH) == aCalendar.get(Calendar.MONTH));
+
+    }
+
+    @Test
+    public void testSendStatement() {
+        AccountVO accountVO = new AccountVO();
+        accountVO.setId(2L);
+        accountVO.setDomainId(1L);
+        Mockito.when(accountDao.findById(Mockito.anyLong())).thenReturn(accountVO);
+
+        QuotaAccountVO acc = new QuotaAccountVO(2L);
+        acc.setQuotaBalance(new BigDecimal(404));
+        acc.setLastStatementDate(null);
+        List<QuotaAccountVO> accounts = new ArrayList<>();
+        accounts.add(acc);
+        Mockito.when(quotaAcc.listAllQuotaAccount()).thenReturn(accounts);
+
+        Mockito.when(quotaUsage.findTotalQuotaUsage(Mockito.anyLong(), Mockito.anyLong(), Mockito.anyInt(), Mockito.any(Date.class), Mockito.any(Date.class)))
+                .thenReturn(new BigDecimal(100));
+
+        // call real method on send monthly statement
+        Mockito.doCallRealMethod().when(quotaStatement).sendStatement();
```

Comment:
This test method has no asserts to verify the results of the send statement operation.  How does it verify that the send statement operation succeeded rather than simply not failing.